reactor-core/src/test/java/reactor/util/context/ContextNTest.java

```diff
@@ -296,4 +296,30 @@ public void putAllOfEmpty() {
 		assertThat(put).isSameAs(c);
 	}
 
+	@Test
+	public void putNonNullWithNull() {
+		int expectedSize = c.size();
+		Context put = c.putNonNull("putNonNull", null);
+
+		assertThat(put).isSameAs(c);
+		assertThat(put.size()).isEqualTo(expectedSize);
+	}
+
+	@Test
+	public void putNonNullWithValue() {
+		int expectedSize = c.size() + 1;
+		Context put = c.putNonNull("putNonNull", "value");
+
+		assertThat(put).isNotSameAs(c);
+		assertThat(put.getOrEmpty("putNonNull")).contains("value");
+		assertThat(put.size()).isEqualTo(expectedSize);
+	}
+
+	@Test
+	public void contextSize() { //renamed to allow import of size() Condition from ContextTest
+		assertThat(c.size()).isEqualTo(6);
+
+		assertThat(c.put("sizeGrows", "yes").size()).isEqualTo(7);
```

Comment:
I would suggest `assertThat(c.put("sizeGrows", "yes")).hasSize(7)`. It will print the resulting Map if assertion fails and help to debug